lib/iris/common/metadata.py

```diff
@@ -1468,12 +1435,52 @@ def values(self):
     if cls is CubeMetadata:
         namespace["_names"] = cls._names
 
-    # Dynamically create the class.
-    Metadata = type(name, bases, namespace)
-    # Now manufacture an instance of that class.
-    metadata = Metadata(cls, **kwargs)
+    # Dynamically create the metadata manager class.
+    MetadataManager = type(name, bases, namespace)
+
+    return MetadataManager
+
+
+def metadata_manager_factory(cls, **kwargs):
+    """
+    A class instance factory function responsible for manufacturing
+    metadata instances dynamically at runtime.
+
+    The factory instances returned by the factory are capable of managing
+    their metadata state, which can be proxied by the owning container.
+
+    Args:
+
+    * cls:
+        A subclass of :class:`~iris.common.metadata.BaseMetadata`, defining
+        the metadata to be managed.
+
+    Kwargs:
+
+    * kwargs:
+        Initial values for the manufactured metadata instance. Unspecified
+        fields will default to a value of 'None'.
+
+    Returns:
+        A manager instance for the provided metadata ``cls``.
+
+    """
+    # Check whether kwargs have valid fields for the specified metadata.
+    if kwargs:
+        extra = [field for field in kwargs.keys() if field not in cls._fields]
+        if extra:
+            bad = ", ".join(map(lambda field: "{!r}".format(field), extra))
+            emsg = "Invalid {!r} field parameters, got {}."
+            raise ValueError(emsg.format(cls.__name__, bad))
```

Comment:
I think it would be cleaner to do this error handling in `Manager.__init__`, at the same time as assigning them, eg
```python
for field in cls._fields:
    setattr(self, field, kwargs.pop(field, None))

if kwargs:
    bad = ", ".join(kwargs.keys())
    raise TypeError(f"Invalid {type(self)!r} parameters: {bad}")
```